r/tests/testthat/test-json.R

```diff
@@ -77,8 +77,8 @@ test_that("Can read json file with nested columns (ARROW-5503)", {
   writeLines('
     { "hello": 3.5, "world": false, "yo": "thing", "arr": [1, 2, 3], "nuf": {} }
     { "hello": 3.25, "world": null, "arr": [2], "nuf": null }
-    { "hello": 3.125, "world": null, "yo": "\u5fcd", "arr": [], "nuf": { "ps": 78 } }
-    { "hello": 0.0, "world": true, "yo": null, "arr": null, "nuf": { "ps": 90 } }
+    { "hello": 3.125, "world": null, "yo": "\u5fcd", "arr": [], "nuf": { "ps": 78.0 } }
+    { "hello": 0.0, "world": true, "yo": null, "arr": null, "nuf": { "ps": 90.0 } }
```

Comment:
What about omitting fields, e.g. `{"hello":"hi"}`.

Reply:
``` r
library(arrow, warn.conflicts = FALSE)
library(tibble)

tf <- tempfile()
writeLines('
    { "hello": 3.5, "world": false, "yo": "thing" }
    { "hello": 2.3}
  ', tf)

read_json_arrow(tf, as_tibble = TRUE)
#> # A tibble: 2 x 3
#>   hello world yo   
#>   <dbl> <lgl> <chr>
#> 1   3.5 FALSE thing
#> 2   2.3 NA    <NA>
```

Maybe this `read_json_arrow()` needs more tests. 

``` r
library(arrow, warn.conflicts = FALSE)
library(tibble)

tf <- tempfile()
writeLines('
    { "hello": 3.5, "world": false, "yo": "thing" }
    { "hello": "hi"}
  ', tf)

read_json_arrow(tf, as_tibble = TRUE)
#> Error in json___TableReader__Read(self): Invalid: Empty JSON file
```

but it's not really this pull request territory. I'm using this at the moment because we currently don't have an R way to create list arrays or struct arrays, i.e. we need to be able to go the other direction with e.g. 

``` r
library(arrow, warn.conflicts = FALSE)
library(tibble)

array(list(1:3, 4:5), type = list_of(int32()))
#> Error in Array__from_vector(x, type): NotImplemented: type not implemented
```

which is the purpose of this one: https://issues.apache.org/jira/browse/ARROW-3809?filter=12344983

Reply:
This is an error of mine, I meant `{"hello": 1.2}` not a bad type. But good if it uncovered an error :)